core/src/operator.rs

```diff
@@ -1561,9 +1561,29 @@ where
 
         #[cfg(test)]
         {
+            use rand::Rng;
```

Comment:
I would prefer to have as little test code as possible in our production code. Can we move this logic into the testparams impls, and call with `self.config.test_params.maybe_corrupt_asserts(&mut asserts)`

Reply:
same for below?